java-squid/src/main/java/org/sonar/java/se/VariableLogicalRelation.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se;
+
+import javax.annotation.CheckForNull;
+
+public abstract class VariableLogicalRelation extends SymbolicValueRelation {
+
+  protected VariableLogicalRelation(SymbolicValue v1, SymbolicValue v2) {
+    super(v1, v2);
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder buffer = new StringBuilder();
+    buffer.append(v1);
+    buffer.append(' ');
+    buffer.append(getOperand());
+    buffer.append(' ');
+    buffer.append(v2);
+    return buffer.toString();
+  }
+
+  protected abstract String getOperand();
+
+  @Override
+  @CheckForNull
+  protected Boolean isImpliedBy(SymbolicValueRelation relation) {
+    if (relation instanceof VariableLogicalRelation) {
+      return isImpliedBySymbolicValueRelation((VariableLogicalRelation) relation);
+    }
+    return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Null is returned but a "Boolean" is expected. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2447)